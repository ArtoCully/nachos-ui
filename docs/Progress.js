import React, { Component } from 'react'
import { View } from 'react-native'

import Progress from '../src/Progress'
import H4 from '../src/typography/H4'
import H5 from '../src/typography/H5'

// WEBPACK (ProgressExample)
class ProgressExample extends Component {
  state = {
    progressValue: 0.5,
  }

  componentDidMount() {
    this._interval = setInterval(
      () => {
        if (this.state.progressValue >= 1) {
          return this.setState({ progressValue: 0 })
        }
        this.setState({
          progressValue: this.state.progressValue + 0.01,
        })
      },
      80
    )
  }

  componentWillUnmount() {
    clearInterval(this._interval)
  }

  render() {
    const progressStyle = { margin: 15 }
    return (
      <View>
        <H4>Example:</H4>
        <Progress
          progress={this.state.progressValue}
          style={progressStyle}
        />
        <H5>
          Progress: {Math.round(this.state.progressValue * 100)}
        </H5>
      </View>
    )
  }
}
// WEBPACK

const description = `
| Prop | Type | Default | Description |
|------|------|---------|-------------|
| \`progress\` | \`number\` | \`0\` | Specifies a progress |
| \`style\` | \`object, array\` | \`void\` | Style the Progress compoment |
| \`width\` | \`number\` | \`300\` | Determines a width of the Progress compoment |
| \`height\` | \`number\` | \`6\` | Determines a height of the Progress compoment |
| \`color\` | \`string\` | \`#2f8cff\` | Specifies a color of the progress line |
`

ProgressExample.styleguide = {
  ...ProgressExample.styleguide,
  title: 'Progress',
  description,
}

export default ProgressExample
