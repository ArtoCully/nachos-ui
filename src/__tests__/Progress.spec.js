import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import Progress from '../Progress'

it('should render correctly', () => {
  const component = renderer.create(<Progress progress={0.5} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
