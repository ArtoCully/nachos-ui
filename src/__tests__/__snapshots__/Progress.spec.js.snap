exports[`test should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#bdc1cc",
      "borderRadius": 3,
      "height": 6,
      "overflow": "hidden",
      "width": 300,
    }
  }>
  <View
    style={
      Object {
        "backgroundColor": "#2f8cff",
        "borderRadius": 3,
        "height": 6,
        "width": 150,
      }
    } />
</View>
`;
